performance_manager/lib/postgres_utils.py

```diff
@@ -16,7 +17,47 @@
 HERE = os.path.dirname(os.path.abspath(__file__))
 
 
-def postgres_create_modified_trigger(
+def get_db_password() -> str:
+    """
+    function to provide rds password
+
+    used to refresh auth token, if required
+    """
+    db_password = os.environ.get("DB_PASSWORD", None)
+    db_host = os.environ.get("DB_HOST")
+    db_port = os.environ.get("DB_PORT")
+    db_user = os.environ.get("DB_USER")
+    db_region = os.environ.get("DB_REGION", None)
+
+    if db_password is None:
+        # generate aws db auth token
+        client = boto3.client("rds")
+        return urlparse.quote_plus(
+            client.generate_db_auth_token(
+                DBHostname=db_host,
+                Port=db_port,
+                DBUsername=db_user,
+                Region=db_region,
+            )
+        )
+
+    return db_password
+
+
+def postgres_event_update_db_password(
+    _: sa.engine.interfaces.Dialect,
+    __: Any,
+    ___: Tuple[Any, ...],
+    cparams: Dict[str, Any],
+) -> None:
+    """
+    update database passord on every new connection attempt
+    this will refresh db auth token passwords
+    """
+    cparams["password"] = get_db_password()
```

Comment:
this is a function that hooks up the the sqlalchemy event listening system,

before any new connection to the RDS, this will be called and the password will be updated from the `get_db_password` function